src/libs/Notification/PushNotification/shouldShowPushNotification.ts

```diff
@@ -14,12 +14,17 @@ export default function shouldShowPushNotification(pushPayload: PushPayload): bo
 
     // The payload is string encoded on Android
     if (typeof payload === 'string') {
-        payload = JSON.parse(payload) as string;
+        try {
+            payload = JSON.parse(payload) as JsonObject;
+        } catch {
+            Log.hmmm(`[PushNotification] Failed to parse the payload`, payload);
+            payload = undefined;
+        }
     }
 
-    const data = payload as PushNotificationData;
+    const data = payload ? (payload as PushNotificationData) : undefined;
 
-    if (!data.reportID) {
+    if (!data?.reportID) {
```

Comment:
```suggestion
    if (data?.reportID === undefined) {
```
To account for `reportID=0` (same as in other places)